src/crimson/os/seastore/omap_manager/btree/btree_omap_manager.cc

```diff
@@ -198,13 +198,17 @@ BtreeOMapManager::omap_rm_key_range(
   LOG_PREFIX(BtreeOMapManager::omap_rm_key_range);
   DEBUGT("{} ~ {}", t, first, last);
   assert(first <= last);
-  return omap_list(
-    omap_root,
-    t,
-    first,
-    last,
-    config
-  ).si_then([this, &omap_root, &t](auto results) {
+  return seastar::do_with(
+    std::make_optional<std::string>(first),
+    std::make_optional<std::string>(last),
```

Comment:
I think we don't need wrap first and last with do_with here, referring to the interface comments: https://github.com/ceph/ceph/blob/b66b8fde2e1cea49a8d0e0c1109092f0c378952c/src/crimson/os/seastore/omap_manager.h#L178-L179

Also, the caller has already extended the lifecycle of first and last correctly: https://github.com/ceph/ceph/blob/b66b8fde2e1cea49a8d0e0c1109092f0c378952c/src/crimson/os/seastore/seastore.cc#L1491-L1511

Reply:
> I think we don't need wrap first and last with do_with here, referring to the interface comments:
> 
> https://github.com/ceph/ceph/blob/b66b8fde2e1cea49a8d0e0c1109092f0c378952c/src/crimson/os/seastore/omap_manager.h#L178-L179
> 
> Also, the caller has already extended the lifecycle of first and last correctly:
> 
> https://github.com/ceph/ceph/blob/b66b8fde2e1cea49a8d0e0c1109092f0c378952c/src/crimson/os/seastore/seastore.cc#L1491-L1511

@cyx1231st Not really, `BtreeOMapManager::omap_list()` is accepting `const std::optional<std::string> &`, so actually if we don't wrap the invocation to `BtreeOMapManager::omap_list()` here, the invocation would create a new `std::optional<std::string>` object that is alive only before the invocation is finished. However, continuations created by `BtreeOMapManager::omap_list()` will outlive those invocation, which led to a daggling reference issue.

Reply:
Ah, I'm missing the fact that the types are actually different.

Probably we can drop the interface comments `, must alive during the call`, and also drop the unnecessary `do_with` for `first` and `last` from `SeaStore::_omap_rmkeyrange()` in this commit?

Reply:
It seems that the following code in `SeaStore::_omap_rmkeyrange()` is also capturing `first` and `last` by reference, so maybe we still need those comments.

https://github.com/ceph/ceph/blob/b66b8fde2e1cea49a8d0e0c1109092f0c378952c/src/crimson/os/seastore/seastore.cc#L1511-L1516